econsa/tests/test_copula.py

```diff
@@ -0,0 +1,97 @@
+"""Tests for copula sampling.
+
+This module contains tests for the copula sampling functions.
+
+Note that we are using simple covariance matrix here since calculating sigma then covariance matrix
+causes numerical instability on higher dimension.
+"""
+import chaospy as cp
+import numpy as np
+import pytest
+from scipy.stats import multivariate_normal as multivariate_norm
+
+from econsa.copula import cond_gaussian_copula
+from econsa.sampling import cond_mvn
+
+
+def get_strategies(name):
+    dim = np.random.randint(2, 10)
+
+    full = list(range(0, dim))
+    given_ind = full[:]
+
+    # dependent_n = 1
+    dependent_n = np.random.randint(low=1, high=dim)
+    dependent_ind = np.random.choice(full, replace=False, size=dependent_n)
+
+    for i in dependent_ind:
+        given_ind.remove(i)
+
+    means = np.random.uniform(-100, 100, dim)
+
+    # Draw new cov until is well-conditioned
+    while True:
+        cov = np.identity(dim) * np.random.uniform(0.05, 10)
```

Comment:
You can move this out of the loop, that matrix is always well-conditioned enough.

Reply:
Ok, cool!